src/nvim/tui/tui.c

```diff
@@ -202,18 +126,94 @@ static void tui_stop(UI *ui)
   }
   xfree(data->write_loop);
   unibi_destroy(data->ut);
-  destroy_screen(data);
+  ugrid_free(&data->grid);
+  data->stop = true;
```

Comment:
If I use `<CTRL-z>`, `tui_stop()` is called and I expected that `tui_main()` would exit at once (thread ends).
For example, I don't see the message with the following patch until I use `fg`:
```diff
diff --git a/src/nvim/tui/tui.c b/src/nvim/tui/tui.c
index f678428..63a359e 100644
--- a/src/nvim/tui/tui.c
+++ b/src/nvim/tui/tui.c
@@ -189,6 +189,7 @@ static void tui_main(UIBridgeData *bridge, UI *ui)
   while (!data->stop) {
     loop_poll_events(&tui_loop, -1);
   }
+  fprintf(stderr,"tui_main ends\n");
 
   loop_close(&tui_loop);
   xfree(data);
```

Also program crashes when `fg` is used (here patch is not included).
```c
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f744a839a38 in raise () from /lib64/libc.so.6
(gdb) bt
#0  0x00007f744a839a38 in raise () from /lib64/libc.so.6
#1  0x00007f744a83ae0a in abort () from /lib64/libc.so.6
#2  0x000000000062abc7 in uv_mutex_lock (mutex=<optimized out>)
    at /home/oni-link/git/neovim/.deps/build/src/libuv/src/unix/thread.c:120
#3  0x000000000061d0bb in loop_schedule (loop=0x7f7449c528c0, event=...) at /home/oni-link/git/neovim/src/nvim/event/loop.c:62
#4  0x0000000000618890 in tui_scheduler (event=..., d=0x7f744a03a2c0) at /home/oni-link/git/neovim/src/nvim/tui/tui.c:202
#5  0x00000000004d8a8d in ui_bridge_put (b=0x7f744a024a80, text=0x7ffc4f5af790 " ", size=1)
    at /home/oni-link/git/neovim/src/nvim/ui_bridge.c:244
#6  0x00000000005d0f44 in send_output (ptr=0x7ffc4f5af760) at /home/oni-link/git/neovim/src/nvim/ui.c:353
#7  0x00000000005d0d46 in ui_puts (str=0x7ffc4f5af790 " ") at /home/oni-link/git/neovim/src/nvim/ui.c:306
#8  0x00000000005d0da0 in ui_putc (c=32 ' ') at /home/oni-link/git/neovim/src/nvim/ui.c:314
#9  0x000000000049beba in screen_char (off=1223, row=8, col=135) at /home/oni-link/git/neovim/src/nvim/screen.c:5732
#10 0x000000000049c3c4 in screen_fill (start_row=1, end_row=68, start_col=0, end_col=136, c1=126, c2=32, attr=6)
    at /home/oni-link/git/neovim/src/nvim/screen.c:5865
#11 0x0000000000490b2b in win_draw_end (wp=0x7f744a04e000, c1=126, c2=32, row=1, endrow=68, hl=HLF_EOB)
    at /home/oni-link/git/neovim/src/nvim/screen.c:1648
#12 0x00000000004904c1 in win_update (wp=0x7f744a04e000) at /home/oni-link/git/neovim/src/nvim/screen.c:1527
#13 0x000000000048d5d4 in update_screen (type=40) at /home/oni-link/git/neovim/src/nvim/screen.c:419
#14 0x000000000049f972 in screen_resize (width=136, height=70) at /home/oni-link/git/neovim/src/nvim/screen.c:7279
#15 0x00000000005d04fd in ui_refresh () at /home/oni-link/git/neovim/src/nvim/ui.c:142
#16 0x00000000005ba2dc in ex_stop (eap=0x7ffc4f5afb60) at /home/oni-link/git/neovim/src/nvim/ex_docmd.c:5937
#17 0x00000000005b1a5c in do_one_cmd (cmdlinep=0x7ffc4f5afc80, sourcing=1, cstack=0x7ffc4f5afd90, fgetline=0x0, cookie=0x0)
    at /home/oni-link/git/neovim/src/nvim/ex_docmd.c:2195
#18 0x00000000005ae5df in do_cmdline (cmdline=0x69fd5e "st", fgetline=0x0, cookie=0x0, flags=11)
    at /home/oni-link/git/neovim/src/nvim/ex_docmd.c:621
#19 0x00000000005adc5d in do_cmdline_cmd (cmd=0x69fd5e "st") at /home/oni-link/git/neovim/src/nvim/ex_docmd.c:293
#20 0x00000000005661ee in nv_suspend (cap=0x7ffc4f5b0300) at /home/oni-link/git/neovim/src/nvim/normal.c:6011
#21 0x000000000055a9b5 in normal_cmd (oap=0x7ffc4f5b03a0, toplevel=true) at /home/oni-link/git/neovim/src/nvim/normal.c:912
#22 0x0000000000553227 in main_loop (cmdwin=0, noexmode=0) at /home/oni-link/git/neovim/src/nvim/main.c:747
#23 0x0000000000552be0 in main (argc=3, argv=0x7ffc4f5b0668) at /home/oni-link/git/neovim/src/nvim/main.c:534
```